src/sidebar.c

```diff
@@ -191,7 +191,7 @@ void sidebar_update_tag_list(GeanyDocument *doc, gboolean update)
 
 	g_return_if_fail(doc == NULL || doc->is_valid);
 
-	if (update)
+	if (update && doc != NULL)
```

Comment:
Definitely good to add, this code was actually written by me and I think I just fell into the misunderstanding of the `g_return_if_fail()` trap above.